butterfly-core/src/main/java/com/paypal/butterfly/core/UpgradePath.java

```diff
@@ -59,8 +60,12 @@
         }
         if (upgradeVersion == null || upgradeVersion.isEmpty()) {
             upgradeVersion = getLastVersion();
+        } else if(firstStep.getCurrentVersion().equals(upgradeVersion)){
+            throw new IllegalArgumentException("The requested upgrade version is the same as the version the application is currently at");
+        } else if(upgradeVersionIsOlder(upgradeVersion)){
+            throw new IllegalArgumentException("The requested upgrade version is older than the version the application is currently at");
         } else if(!upgradeVersionValidation(upgradeVersion)) {
-            throw new IllegalArgumentException("Upgrade version " + upgradeVersion + " is invalid");
+            throw new IllegalArgumentException("The requested upgraded version is inexistent");
```

Comment:
Let's add the version in the exception message please. Like this:

`throw new IllegalArgumentException("The requested upgraded version (" + upgradeVersion + ") is inexistent");`

Reply:
Made necessary changes.